scripts/backup_reduction_scripts.py

```diff
@@ -0,0 +1,150 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+This script is expected to be called by a cronjob daily.
+
+It will traverse all instruments in VALID_INSTRUMENTS, grab reduce.py and reduce_vars.py
+and upload it to the repository.
+
+The repository of the STORAGE_DIR needs to be manually configured to point to the correct
+remote - otherwise this script will fail to commit/push.
+
+There is also a --dry-run option that will just show which files are traversed,
+but will not do anything.
+
+The cronjob should be:
+
+0 1 * * * python3 /home/reduce/backup_reduction_scripts.py
+
+The reason it runs at 1AM is that the queue_processor restarting happens at midnight and
+there might be some interference. There shouldn't be, but just in case it doesn't hurt
+to be scheduled for a bit later.
+"""
+import os
+import sys
+import argparse
+import datetime
+import logging
+import shutil
+import traceback
+from pathlib import Path
+
+import git
+from git import Git
+
+from utils.project.static_content import LOG_FORMAT
+from utils.project.structure import get_log_file
+
+ISIS_MOUNT_PATH = Path("/isis")
+AUTOREDUCTION_PATH = Path("user/scripts/autoreduction")
+REDUCE_FILES_TO_SAVE = ["reduce.py", "reduce_vars.py"]
+
+STORAGE_DIR = Path("~/autoreduction_scripts").expanduser().absolute()
```

Comment:
```suggestion
# STORAGE_DIR is the git repository dir that has been configured to point to the correct remote
STORAGE_DIR = Path("~/autoreduction_scripts").expanduser().absolute()
```

Reply:
An extra comment to help, and where someone has not read the script intro blurb in detail